src/expr/src/scalar/func.rs

```diff
@@ -2825,6 +2825,42 @@ fn text_concat_variadic<'a>(datums: &[Datum<'a>], temp_storage: &'a RowArena) ->
     Datum::String(temp_storage.push_string(buf))
 }
 
+fn pad_leading<'a>(
+    datums: &[Datum<'a>],
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let string = datums[0].unwrap_str();
+
+    let len = match usize::try_from(datums[1].unwrap_int64()) {
+        Ok(len) => len,
+        Err(_) => {
+            return Err(EvalError::InvalidParameterValue(
+                "length must be positive".to_owned(),
+            ))
+        }
+    };
+
+    let pad_string = if datums.len() == 3 {
+        datums[2].unwrap_str()
+    } else {
+        " "
+    };
+
+    let mut buf: String;
+    if len < string.len() {
```

Comment:
Note that in rust, string.len() returns the length in bytes and not characters. See: https://doc.rust-lang.org/std/string/struct.String.html#method.len